source/iostreams.tex

```diff
@@ -728,19 +728,7 @@
 integral types.
 It is used to represent the number of characters transferred in an I/O
 operation, or the size of I/O buffers.\footnote{\tcode{streamsize}
-is used in most places where ISO C would use
-\tcode{size_t}.
-Most of the uses of
-\tcode{streamsize}
-could use
-\tcode{size_t},
-except for the
-\tcode{strstreambuf}
-constructors, which require negative values.
-It should probably be the signed type corresponding to
-\tcode{size_t}
-(which is what Posix.2 calls
-\tcode{ssize_t}).}
+is used in most places where ISO C would use \tcode{size_t}.}
```

Comment:
Sorry I've only noticed this now, but removing the mention of `ssize_t` seems unfortunate. Also, what remains of the footnote is a lie. We don't use `streamsize` for the size of an array _new-expression_ or the size passed to `operator new` (corresponding to `malloc` in C), or the length passed to `char_traits::compare` (`memcmp`), or "most" other places. Maybe we should qualify this footnote to be specific to I/O in C, or just to places that actually use `streamsize` e.g.

> Most places where `streamsize` is used would use `size_t` in ISO C, or `ssize_t` in POSIX.

Reply:
Could you make a PR so we don't lose track of this? The suggestion sounds good.

Reply:
Incorporated into PR #4299